google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java

```diff
@@ -351,15 +351,42 @@ public final MediaHttpUploader getMediaHttpUploader() {
    */
   protected final void initializeMediaUpload(AbstractInputStreamContent mediaContent) {
     HttpRequestFactory requestFactory = abstractGoogleClient.getRequestFactory();
+    String applicationName = abstractGoogleClient.getApplicationName();
+    HttpRequestInitializer requestInitializer =
+        mediaUploadRequestUserAgentInitializer(applicationName, requestFactory.getInitializer());
     this.uploader =
```

Comment:
Special handling for `applicationName`,
<img width="744" alt="Screenshot 2023-01-20 at 9 49 39 AM" src="https://user-images.githubusercontent.com/28604/213727558-dfd21559-8d70-44bb-9808-6739eaebce69.png">